### PR TITLE
test: add comprehensive tests for Stream functionality

### LINKED ISSUE #661: feat: Add Unit Tests for stream.go
#### **Problem**

The OLake codebase currently has unit tests for:

* `resolver`
* `flattener`
* `catalog`

However, the file `olake/types/stream.go` does not have any test coverage.

This file handles critical stream definition and configuration logic for the OLake connector.

Without unit tests, bugs in stream construction, incorrect destination name generation, builder method chaining issues, schema field management, JSON deserialization edge cases, or utility function failures may slip through during development. This is particularly critical as the `Stream` struct is used extensively across the codebase.

---

#### **Proposed Solution**

Add comprehensive unit tests for the logic inside `stream.go`.

This will help:

* Increase overall test coverage
* Make contributions safer when modifying stream-related logic

For reference on how existing unit tests are structured, you can refer to **PR #273** and the existing `catalog_test.go` file.

---

#### **Notes for Contributors**

* Add tests covering most of the scenarios and edge cases which should indicate both pass and fail scenarios.

* Follow the existing test file structure and naming conventions to maintain consistency with other parts (see `catalog_test.go` for reference).

* Use `testify/assert` and `testify/require` for assertions (as seen in existing test files).

### PR DESCRIPTION
**Pull request by Beeram12**
# Description

Fixes #661

This PR adds comprehensive unit tests for the `stream.go` file logic. The tests cover the `Stream` struct and its associated helper functions, following the table-driven test pattern with multiple edge cases.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] `TestStream_NewStream` - Checks if a new stream is created correctly with all necessary fields initialized.
- [x] `TestStream_WithSyncMode` - Checks if helper method correctly adds sync modes with edge cases.
- [x] `TestStream_WithPrimaryKey` - Checks if helper method correctly adds primary keys with edge cases.
- [x] `TestStream_WithCursorField` - Checks if helper method correctly adds cursor fields with edge cases.
- [x] `TestStream_WithSchema` - Checks if schema assignment works correctly.
- [x] `TestStream_Wrap` - Checks if stream wrapping works with different index values.
- [x] `TestStream_WithUpsertField` - Checks if adding fields to the schema works for different data types.
- [x] `TestStream_UnmarshalJSON` - Checks if the system handles partial or empty JSON data safely without crashing.
- [x] `TestStreamsToMap` - Checks if streams are correctly converted to a map with proper key-value mapping.
- [x] `TestLogCatalog` - Checks if the catalog file is actually written to the disk with the correct data.

All tests follow the table-driven test pattern with comprehensive edge cases and are verified by running:
`go test ./types -v` or 
for specific tests:
`go test ./types -v -run TestStream_`

# Screenshots or Recordings
https://github.com/user-attachments/assets/3d40ecb5-15a6-44ff-8ef5-bdd0cde95f5a

### Video Content:
- Running specific test: `go test ./types -v -run TestStream_`
- Test output showing all tests passing

## Documentation

- [ ] Documentation Link: [link to README, olake.io/docs, or olake-docs]
- [x] N/A (bug fix, refactor, or test changes only)

## Related PR's (If Any):
N/A

### CODE REVIEW

**Comment by Beeram12** [comment]
@nayanj98  Hey, I think it took a lot of time. I was busy with some other work. Let me know if there are any changes to make

**Comment by nayanj98** [comment]
@Beeram12 can you please rebase your PR to staging.

**Comment by Beeram12** [comment]
@nayanj98 does this look good?

**Comment by nayanj98** [comment]
@Beeram12 Perfect! I will assign someone to review your work soon

**Comment by nayanj98** [comment]
@Beeram12 I have assigned @vaibhav-datazip to review your work

File: types/stream_test.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	assert.Equal(t, "public", stream.Namespace)
+
+	assert.NotNil(t, stream.SupportedSyncModes, "SupportedSyncModes should be initialized")
+	assert.NotNil(t, stream.SourceDefinedPrimaryKey, "SourceDefinedPrimaryKey should be initialized")
+	assert.NotNil(t, stream.AvailableCursorFields, "AvailableCursorFields should be intialized")
+	assert.NotNil(t, stream.Schema, "Schema is not defined")
+
+	assert.NotEmpty(t, stream.DestinationDatabase, "DestinationDatabase should be generated")
+	assert.NotEmpty(t, stream.DestinationTable, "DestinationTable should be generated")
+}
```
Comment (vaibhav-datazip) [inline types/stream_test.go]: you can check the function naming convention for test files from existing like stream_configured_test.go and change the name accordingly in order to maintain consistency.

File: types/stream_test.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+func TestWithCursorField(t *testing.T) {
+	stream := NewStream("users", "public", nil)
+
+	outputStream := stream.WithCursorField("updated_at", "inserted_at")
+
+	assert.Same(t, stream, outputStream, "Should return the same instance")
+
+	assert.True(t, stream.AvailableCursorFields.Exists("updated_at"), "Should contain 'updated_at'")
```
Comment (vaibhav-datazip) [inline types/stream_test.go]: also in other functions as well you can add more edge cases, and name these test cases in array and iterate using a for loop to assert. like we have in other unit test files as well

File: types/stream_test.go (hunk 1 above)
Comment (Beeram12) [reply types/stream_test.go]: I have changed the naming convention accordingly

File: types/stream_test.go (hunk 2 above)
Comment (Beeram12) [reply types/stream_test.go]: added more test cases like done in other unit test files

**Comment by vaibhav-datazip** [comment]
@Beeram12  Also, to understand how the stream.go file is used by the rest of the codebase, please run the sync once and observe the edge cases that get triggered. This will help you reason about the edge cases more effectively.

You can refer to the [Olake commands and flags documentation](https://olake.io/docs/community/commands-and-flags/) to understand the different places where the stream file can be passed or manipulated.

Please attach a short video of you running the sync here as well. This will help me understand how familiar you are with the flow, and I can then review any edge cases that might have been missed and let you know.

Also please mark this PR as ready for review when you have done change and want this to be reviewed again

**Comment by vaibhav-datazip** [comment]
Hi @Beeram12 , you haven't replied for nearly a week, please do if you have made any progress or have any doubt. Otherwise we will consider you inactive and close this PR.

**Comment by vaibhav-datazip** [comment]
Hi @Beeram12 , closing this pr as there was no response from your side

**Comment by Beeram12** [comment]
> Hi @Beeram12 , closing this pr as there was no response from your side

Hi vaibhav I am actually busy with other works so I am unable to focus on this can I just open the Pr once I get free if you don't mind. I am interested in working but I am having time constraints

**Comment by vaibhav-datazip** [comment]
> Hi vaibhav I am actually busy with other works so I am unable to focus on this can I just open the Pr once I get free if you don't mind. I am interested in working but I am having time constraints

Yeah sure, please feel free to reopen it and if you face any problem you can tag me here or on slack
